ReSharper.FSharp/src/FSharp.Psi.Intentions/src/Intentions/UseNamedAccessAction.fs

```diff
@@ -0,0 +1,107 @@
+open JetBrains.ReSharper.Plugins.FSharp.Psi
+
+[<ContextAction(Name = "UseNamedAccess", Group = "F#", Description = "Use named access inside a DU pattern")>]
+type UseNamedAccessAction(dataProvider: FSharpContextActionDataProvider) =
+    inherit FSharpContextActionBase(dataProvider)
+    
+    let mutable names = Array.empty
+    let mutable tuplePat : ITuplePat = null
+
+    override x.Text = "Use named access"
+
+    override x.IsAvailable _ =
+        let pattern = dataProvider.GetSelectedElement<IParametersOwnerPat>()
+        if isNull pattern then false else
+        // We expect a single IParenPat here
+        if pattern.Parameters.Count <> 1 then false else
+        let parenPat =
+            match pattern.Parameters.[0] with
+            | :? IParenPat as parenPat -> parenPat
+            | _ -> null
+
+        if isNull parenPat then false else
+        // This should only work when there are multiple items
+        tuplePat <- parenPat.Pattern :?> ITuplePat
+        if isNull tuplePat then false else
+        
+        let hasUnderscore =
+            tuplePat.Patterns
+            |> Seq.exists (fun p ->
+                match p with
+                | :? IWildPat -> true
+                | _ -> false)
+    
+        // We need something that is currently ignored
+        if not hasUnderscore then false else
+
+        // We need to be sure that the pattern is a DU with all named fields
+        let fcsUnionCase = pattern.ReferenceName.Reference.GetFcsSymbol().As<FSharpUnionCase>()
+        if isNull fcsUnionCase then false else
+            
+        names <-
+            fcsUnionCase.Fields
+            |> Seq.choose (fun field -> if field.IsNameGenerated then None else Some field.Name)
+            |> Seq.toArray
+        
+        // All fields need to be named
+        if names.Length <> fcsUnionCase.Fields.Count then false else
+        // The amount of fields needs to match with the tuple length
+        if names.Length <> tuplePat.Patterns.Count then false else
+
+        true
+
+    override x.ExecutePsiTransaction(_,_) =
+        let pattern = dataProvider.GetSelectedElement<IParametersOwnerPat>()
+        if isNull pattern then null else
+        use writeCookie = WriteLockCookie.Create(pattern.IsPhysical())
+        use disableFormatter = new DisableCodeFormatter()
+        
+        let usedFieldsWithPatterns: (string * IFSharpPattern)[] =
+            (names, tuplePat.PatternsEnumerable)
+            ||> Seq.zip
+            |> Seq.choose (fun (name, pat) ->
+                match pat with
+                | :? IWildPat -> None
+                | _ -> Some (name, pat))
+            |> Seq.toArray
+        
+        let factory = pattern.CreateElementFactory()
+        let sourceText =
+            let fields =
+                usedFieldsWithPatterns
+                |> Array.map (fun (name,_) -> $"{name} = _")
+                |> String.concat "; "
+            $"{pattern.Identifier.Name}({fields})"
```

Comment:
This may in theory break multiline parameter owner patterns or multiline field patterns. The tree would be fixed automatically if we had R# formatter finished, but for now we could, perhaps, disable the action on multiline patterns.

Reply:
Added an extra check to not make the intention available when the paren pattern is multiline.